src/lib/components/dialog/QDialog.svelte

```diff
@@ -19,41 +18,51 @@
   const emit = createEventDispatcher();
   let dialogElement: HTMLDialogElement;
 
-  $: canHideOnClickOutside = value === true && persistent !== true;
+  let opened = false;
+
+  onMount(() => {
+    opened = dialogElement.open;
+    opened ? (dialogElement.style.display = "block") : (dialogElement.style.display = "none");
+  });
+
+  $: canHideOnClickOutside = opened && persistent !== true;
 
   $: positionClass = ["top", "right", "bottom", "left"].includes(position!) ? position : undefined;
 
   $: classes = createClasses(
-    [value && "active", positionClass, modal && "modal", fullscreen && "max"],
+    [opened && "active", positionClass, modal && "modal", fullscreen && "max"],
     {
       component: "q-dialog",
       userClasses,
     }
   );
 
-  $: value === true
-    ? modal
-      ? dialogElement?.showModal()
-      : dialogElement?.show()
-    : dialogElement?.close();
-
   export function hide() {
-    if (value === true) {
-      value = false;
+    if (dialogElement && dialogElement.open) {
+      dialogElement.close();
+      opened = false;
+      setTimeout(() => {
+        dialogElement.style.display = "none";
+      }, 250);
     }
   }
   export function show() {
-    if (value === false) {
-      value = true;
+    if (dialogElement && !dialogElement.open) {
+      modal ? dialogElement.showModal() : dialogElement.show();
+      opened = true;
+      dialogElement.style.display = "block";
     }
   }
   export function toggle(e: MouseEvent) {
-    value = !value;
+    if (dialogElement) {
+      opened = !opened;
+      dialogElement.open ? hide() : show();
```

Comment:
Don't use ternary to replace if/else. Ternary should be used for assignments, not function calls.